docs/howto/run_universal_discovery.md

```diff
@@ -0,0 +1,281 @@
+element* will configure the Universal Discovery to only allow *container*
+Content items to be picked.
+
+Example:
+
+```html
+<button type="button" class="ez-js-run-universal-discovery"
+    data-ud-title="Pick a container"
+    data-ud-container>Run UDW</button>
+```
+
+#### Content items of a Content Types set
+
+The `data-ud-content-type-identifiers` can be used to let the user to only pick
+Content items of a Content Types set. This attribute expects a space separated
+list of Content Type identifiers.
+
+Example:
+
+```html
+<button type="button" class="ez-js-run-universal-discovery"
+    data-ud-title="Pick a Folder or an Article"
+    data-ud-content-type-identifiers="folder article">Run UDW</button>
+```
+
+#### Custom restriction
+
+To implement more advanced restrictions, the *run Universal Discovery* mechanism
+also exposes the `ez:runUniversalDiscovery:select` JavaScript event.  This event
+is dispatched from the *run Universal Discovery element* when the user is about
+to pick a Content item/Location. It is configured [to
+*bubble*](https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-flow-bubbling)
+and to [be
+*cancelable*](https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-flow-cancelation).
+Preventing the default behaviour of this event allows to restrict the user from
+picking the corresponding Content item/Location.
+
+<p id="selection-object">The event detail provides a selection object containing:</p>
+
+* the Content item object under `content` as generated by
+  [`eZ.Content#toObject`](https://github.com/ezsystems/PlatformUIBundle/blob/2.0/Resources/public/js/models/ez-contentmodel.js#L468)
+* the Location object under `location` as generated by
+  [`eZ.Location#toObject`](https://github.com/ezsystems/PlatformUIBundle/blob/2.0/Resources/public/js/models/ez-locationmodel.js#L423)
+* the Content Type object under `contentType` as generated by
+  [`eZ.ContentType#toObject`](https://github.com/ezsystems/PlatformUIBundle/blob/2.0/Resources/public/js/models/ez-contenttypemodel.js#L188)
+
+Example: only allow Content items with a Field which identifier is `image`.
+
+```js
+function hasAnImageField(selection) {
+    return (
+        typeof selection.contentType.fieldDefinitions['image'] !== 'undefined'
+    );
+}
+
+document.addEventListener('ez:runUniversalDiscovery:select', function (e) {
+    const selection = e.detail.selection; // selection object
+    // const runUDElement = e.target;
+    // runUDElement might be useful to retrieve the context in which the UD is run
+
+    if ( !hasAnImageField(selection) ) {
+        e.preventDefault();
+    }
+});
+```
+
+### Use the Universal Discovery selection
+
+The *run Universal Discovery* convention allows to handle the Universal
+Discovery selection validated by the user. As for the selection restriction, a
+built-in behavior can be configured and a JavaScript event allows to implement
```

Comment:
built-in behavior can be configured and a JavaScript event allows **you** to implement